### PR TITLE
For B0N: Move boot/validation code (boot_from()) into subsys.

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Add to a new file (bl_boot) and rename the function to
bl_validate_and_boot().

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by hakonfam** [comment]
I really like this. Have you looked at the effort needed to have unit tests for this?

**Comment by oyvindronningstad** [comment]
Should we move the bl_validate_fw call out of bl_boot back into main to have more flexibility?

**Comment by hakonfam** [comment]
> Should we move the bl_validate_fw call out of bl_boot back into main to have more flexibility?

If we look at the 53 network core bootloader I think this makes sense.

File: subsys/bootloader/bl_boot/bl_boot.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		__set_CONTROL(__get_CONTROL() & ~CONTROL_SPSEL_Msk);
+	}
+
+	__DSB(); /* Force Memory Write before continuing */
+	__ISB(); /* Flush and refill pipeline with updated permissions */
+
+	VTOR = fw_info->address;
+	u32_t *vector_table = (u32_t *)fw_info->address;
+
+	if (!fw_info_ext_api_provide(fw_info, true)) {
```
Comment (hakonfam) [inline subsys/bootloader/bl_boot/bl_boot.c]: I guess this will also work fine in the context of nrf53 network core? I will try it.

**Comment by oyvindronningstad** [comment]
@sigvartmh Can you take a look so we can merge this?

**Pull request merged**